test/async-hooks/test-unhandled-rejection-context.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const initHooks = require('./init-hooks');
+const async_hooks = require('async_hooks');
+
+if (!common.isMainThread)
+  common.skip('Worker bootstrapping works differently -> different async IDs');
+
+const promiseAsyncIds = [];
+const hooks = initHooks({
+  oninit(asyncId, type) {
+    if (type === 'PROMISE') {
+      promiseAsyncIds.push(asyncId);
+    }
+  }
+});
+
+hooks.enable();
+
+// Promise.resolve()
+// .then(() => {
+//   throw new Error('Throwing on purpose');
+// });
+
+Promise.reject();
```

Comment:
@benjamingr I noticed an interesting case. If you just do `Promise.reject` the `async_hooks.executionAsyncId()` is incorrect in `process.on('unhandledRejection')`'s callback. But it is correct if you do,

```
Promise.resolve()
.then(() => {
   throw new Error('Throwing on purpose');
 });
```

Therefore, I tried out the setting the context from `node_task_queue`, and then it seems to be correct.

Let me know your thoughts.